security/management/commands/purge_logs.py

```diff
@@ -0,0 +1,145 @@
+import os
+import json
+import gzip
+
+from datetime import datetime, time, timedelta
+
+from six.moves import input
+
+from django.core import serializers
+from django.core.files.base import ContentFile
+from django.core.serializers.json import DjangoJSONEncoder
+from django.core.management.base import CommandError
+from django.core.management.base import BaseCommand
+from django.utils.timezone import now, utc
+from django.utils.module_loading import import_string
+from django.utils.encoding import force_bytes, force_text
+
+from security.config import settings
+from security.models import InputLoggedRequest, OutputLoggedRequest, CommandLog, CeleryTaskLog
+
+from io import BytesIO, TextIOWrapper
+
+
+storage = import_string(settings.BACKUP_STORAGE_CLASS)()
+
+
+UNIT_OPTIONS = {
+    'h': lambda amount: now() - timedelta(hours=amount),
+    'd': lambda amount: now() - timedelta(days=amount),
+    'w': lambda amount: now() - timedelta(weeks=amount),
+    'm': lambda amount: now() - timedelta(days=(30 * amount)),  # 30-day month
+    'y': lambda amount: now() - timedelta(weeks=(52 * amount)),  # 364-day year
+}
+
+
+def lazy_serialize_qs_without_pk(qs):
+    def generator():
+        for obj in qs.iterator():
+            data = serializers.serialize('python', [obj])[0]
+
+            del data['pk']
+            yield data
+
+    class StreamList(list):
+        def __iter__(self):
+            return generator()
+
+        # according to the comment below
+        def __len__(self):
+            return 1
+
+    return StreamList()
```

Comment:
little bit hack, json.dump cannot serialize iterator WTF

Reply:
how the hell did you even come up with this

Reply:
:D